tasks/epi_stats/fetch_citystats.py

```diff
@@ -0,0 +1,86 @@
+import logging
+from datetime import date
+import pandas as pd
+import numpy as np 
+import functools
+import luigi
+from backend.config import get_config
+from backend.repository import AWSFileRepository
+from backend.metrics.calculations import fourteen_day_avg_ratio
+
+log = logging.getLogger(__name__)
+
+def critical_districts(data):
+    """
+    1) Criteria for identifying cities/districts with (a) increasing Covid-19 Incidence:
+    a. Daily new cases  >  100 
+    b. New Cases in the last 14 days  >  New cases in the preceding 14 days
+    AND
+    2) Criteria for identifying cities with high incidence burden:
+    a. Top 20 cities/districts with the highest incidence burden as of date in terms of cumulative cases
+    """
+    # latest date
+    data_latest = data[data['date'] == data['date'].max()]
+
+    #? criteria 1 a 
+    c1a = data_latest['delta.confirmed'] > 100
+    #? criteria 1 b
+    data_latest['total.confirmed.14_day_ratio'] = np.where(abs(data_latest['total.confirmed.14_day_ratio'].values) == np.inf, np.nan, data_latest['total.confirmed.14_day_ratio'].values)
+    c1b = data_latest['total.confirmed.14_day_ratio'] > 1
+    
+    
+    n = 20  #################### TO n, ASSIGN NUMBER OF CITIES SATISFYING CRITERIA 1 TO BE PICKED #########
+    criteria1 = (c1a & c1b)
+    critical_cities_c1 = data_latest[criteria1][['district', 'delta.confirmed']]
+    critical_cities_c1.sort_values(by=['delta.confirmed'], ascending=False, inplace=True)
+    critical_cities_c1_capped = critical_cities_c1.head(n)    
+    
+    # criteria 2
+    top20_cutoff = data_latest["total.confirmed"].sort_values(ascending=False).iloc[20]
```

Comment:
is that 20 supposed to be the "n" variable

Reply:
No the 20 is for criteria 2 (top 20 by cumulative) and n is for criteria 1 (top n by delta.confirmed of those districts satisfying criteria 1) are supposed to be different (i understand the confusion)